### DISCUSSION TITLE
Post UploadFile with extra parameter

### DISCUSSION THREAD

**Discussion by PaulSka** [fastapi/fastapi#8121] [Questions] [answered]
**Description**

I try do post some extra data with a UploadFile, how it's possible ?
FastApi return `Error: Unprocessable Entity`

Example :
```python

class MetadataInfoAdd(BaseModel):
    """
        Metadata Info Add Model
    """
    document: str

class MetadataInfo(BaseModel):
    """
        Metadata Info Model
    """
    metadata_key: str
    metadata_value: str
    metadata_type: str

@app.post("/metadata/add", tags=["stable"], response_model=MetadataInfoAdd)
async def metadata_add(metadata: List[MetadataInfo], file: UploadFile = File(...)):
    """
        Perform POST metadata add
    """
    return {"document" : "STR"}
```
Return this
![0526](https://user-images.githubusercontent.com/1415425/58785612-24547d80-85e6-11e9-81d2-43dec81d6566.png)

I change my async def (remove file parameter) like this :
```python

@app.post("/metadata/add", tags=["stable"], response_model=MetadataInfoAdd)
async def metadata_add(metadata: List[MetadataInfo]):
    """
        Perform POST metadata add
    """
    return {"document" : "STR"}
``` 

It's work
![0527](https://user-images.githubusercontent.com/1415425/58785687-4817c380-85e6-11e9-88aa-d4e7469ce7e8.png)

Is it possible to do this ?

**Comment by tiangolo** [comment, marked as answer]
It's a limitation of HTTP itself. To upload files the client sends the body payload encoded with "form data".

Any additional data can only be sent using form data, so, simple fields.

It's documented here in a warning: https://fastapi.tiangolo.com/tutorial/request-forms-and-files/#define-file-and-form-parameters

**Reply by pythonweb2** [reply]
A string field is a simple field, and can contain JSON. I posted an answer here about how you can send a JSON payload in a form request: #8406.

**Comment by PaulSka** [comment]
Ok thanks for reply :)

**Comment by bharath-kumarn** [comment]
Use like this
**Define File and Form parameters¶
Create file and form parameters the same way you would for Body or Query:**


from fastapi import FastAPI, File, Form, UploadFile

app = FastAPI()


@app.post("/files/")
async def create_file(
    file: bytes = File(...), fileb: UploadFile = File(...), token: str = Form(...)
):
    return {
        "file_size": len(file),
        "token": token,
        "fileb_content_type": fileb.content_type,
    }
